commons/all/src/main/java/org/infinispan/commons/configuration/attributes/Attribute.java

```diff
@@ -216,6 +222,42 @@ public boolean matches(Attribute<?> other) {
       return true;
    }
 
+   /**
+    * Updates this attribute with the value of the other attribute only if the attribute is mutable and the other has
+    * been modified from its default value
+    * @param other
+    */
+   @Override
+   public void update(Attribute<T> other) {
+      if (this.definition.equals(other.definition)) {
+         if (isImmutable()) {
+            // Ensure that there are no incompatible changes
+            if (definition.isGlobal() && !Objects.equals(value, other.value)) {
+               throw Log.CONFIG.incompatibleConfiguration(definition.name(), String.valueOf(value), String.valueOf(other.value));
+            }
+         } else if (other.isModified()) {
```

Comment:
I don't think this will work for the following scenario:

1. User creates a cache configuration containing a non-default value for an attribute that can be updated.
2. User then updates cache configuration so that the default value for said attribute is desired.

Reply:
I've changed it behave like you said. I was thinking of touching only the modified attributes, but it made no sense.
I'm only triggering the attribute changed events if the values are different